dll/win32/shell32/utils.cpp

```diff
@@ -297,3 +297,35 @@ LargeIntegerToString(
     return Int64ToString(pLargeInt->QuadPart, pszOut, cchOut, bUseFormat,
                          pNumberFormat, dwNumberFlags);
 }
+
+/*************************************************************************
+ *                PathIsEqualOrSubFolder (SHELL32.755)
+ */
+EXTERN_C
+BOOL WINAPI PathIsEqualOrSubFolder(_In_ LPCWSTR pszFile1OrCSIDL, _In_ LPCWSTR pszFile2)
+{
+    WCHAR szCommon[MAX_PATH], szFile1[MAX_PATH];
+    HRESULT hr;
+
+    TRACE("%s %s\n", debugstr_w(pszFile1OrCSIDL), debugstr_w(pszFile2));
+
+    if (IS_INTRESOURCE(pszFile1OrCSIDL))
+    {
+        WORD nCSIDL = LOWORD(pszFile1OrCSIDL);
+        SHGetFolderPathW(NULL, (nCSIDL | CSIDL_FLAG_DONT_VERIFY), NULL, 0, szFile1);
```

Comment:
Why a WORD conversion? The 2nd parameter to SHGetFolderPathW is a 32-bit value.
You should use something like PtrToUlong.
```suggestion
        SHGetFolderPathW(NULL, PtrToUlong(pszFile1OrCSIDL) | CSIDL_FLAG_DONT_VERIFY, NULL, 0, szFile1);
```